packages/land/contracts/common/ERC721BaseTokenCommon.sol

```diff
@@ -90,6 +90,18 @@ abstract contract ERC721BaseTokenCommon is IContext, IERC721Upgradeable, IERC721
         (owner, ) = _ownerAndOperatorEnabledOf(id);
     }
 
+    /// @param from sender address
+    /// @param owner owner address of the token
+    /// @param id token id to burn
+    function _burn(address from, address owner, uint256 id) internal {
+        if (from != owner) {
```

Comment:
Something stupid here, why have the 2 arguments if anyway you must have from === owner ?
This is probably the fix that added during the 2019 upgrade ... but that means burnFrom on mainnet always reverts (if from !== owner) even if you are a superoperator

Reply:
From must be the owner (you don't burn from somebody else). The sender can be anybody. The call in polygon land `_burn(_msgSender(), _ownerOf(id), id)` is using this check.